src/components/MemberProfileEdit.tsx

```diff
@@ -1,117 +1,186 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import {
   Box,
   VStack,
   FormControl,
   FormLabel,
   Input,
-  Textarea,
+  Switch,
   Button,
   HStack,
 } from '@chakra-ui/react';
-import { Member } from '../types';
+import { Member, SocialField } from '../types';
 
 interface MemberProfileEditProps {
   member: Member;
   onSave: (profile: Partial<Member>) => void;
 }
 
+const isSocialField = (field: keyof Member): boolean => {
+  return field === 'linkedin' || field === 'github' || field === 'leetcode';
+};
+
 const MemberProfileEdit: React.FC<MemberProfileEditProps> = ({
   member,
   onSave,
 }) => {
-  const [profile, setProfile] = useState<Partial<Member>>(member);
+  const [profile, setProfile] = useState<Partial<Member>>({
+    bio: member.bio,
+    major: member.major,
+    gradDate: member.gradDate,
+    linkedin: member.linkedin,
+    github: member.github,
+    leetcode: member.leetcode,
+    local: member.local,
+  });
 
-  const handleChange = (
-    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
-  ) => {
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const { name, value } = e.target;
-    setProfile((p) => ({ ...p, [name]: value }));
+    setProfile((prevProfile) => ({
+      ...prevProfile,
+      [name]: value,
+    }));
+  };
+
+  const handleSocialFieldChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const [field, key] = e.target.name.split('.');
+
+    if (!isSocialField(field as keyof Member)) return;
+
+    setProfile((prevProfile) => ({
+      ...prevProfile,
+      [field]: {
+        ...((prevProfile[field as keyof Member] as SocialField) || {}),
```

Comment:
Do you need this casting etc?

Reply:
yeah, there might be a cleaner way to do the same thing, but the `keyof Member` is so I can index prevProfile which is type `Partial<Member>`, and the `SocialField` is because spread only works for object types, but many a field of Member is not an object

Reply:
there might be a way to add typing upstream perhaps, but I didn't feel like wrangling with typescript to that extent